projects/reservoir/ks.py

```diff
@@ -116,51 +119,85 @@ def _generate_ic(input_dim, seed=0):
 
 
 def generate_ks_time_series(
-    input_size, domain_size, n_steps, timestep=0.25, seed=0,
+    input_size, domain_size, n_steps, timestep=0.25, seed=0, error_eps=0.0
 ):
     ic = _generate_ic(input_size, seed)
-    return integrate_ks_eqn(ic=ic, domain_size=domain_size, dt=timestep, Nt=n_steps)
+    return integrate_ks_eqn(
+        ic=ic, domain_size=domain_size, dt=timestep, Nt=n_steps, error_eps=error_eps
+    )
 
 
 @dataclass
 class KuramotoSivashinskyConfig:
     """ Generates 1D Kuramoto-Sivashinsky solution time series
     N: number spatial points in final output vectors
     domain_size: periodic domain size
-    timestep: timestep in solver
+    timestep: solver timestep
     spatial_downsampling_factor: optional, if >1 solver will run at this factor
         higher resolution and the final time series is downsampled to N.
         This is useful for stability.
-    time_downsampling_factor: optional, same as above but for the time dimension.
     """
 
     N: int
     domain_size: int
     timestep: float
     spatial_downsampling_factor: int = 1
-    time_downsampling_factor: int = 1
+    error_eps: float = 0.0
     subdomain_output_size: Optional[int] = None
     subdomain_overlap: Optional[int] = None
     subdomain_axis: int = 1
 
-    def generate(self, n_steps: int, seed: int = 0):
+    def generate_from_seed(self, n_steps: int, seed: int = 0):
         """
-            n_steps: Number of timesteps to output. The interval between
-                timesteps output is timestep * time_downsampling_factor
+            n_steps: Number of timesteps to output
             seed: Random seed for initial condition
 
         Returns:
             Time series array with dims (time, x)
         """
-        ks_time_series = generate_ks_time_series(
-            input_size=self.N * self.spatial_downsampling_factor,
-            n_steps=n_steps * self.time_downsampling_factor,
-            seed=seed,
-            domain_size=self.domain_size,
-            timestep=self.timestep,
-        )
-
-        ks_time_series = ks_time_series[:: self.time_downsampling_factor, :]
-        ks_time_series = ks_time_series[:, :: self.spatial_downsampling_factor]
+        ic = _generate_ic(self.N * self.spatial_downsampling_factor, seed=seed)
+        ks_time_series = self.predict(n_steps=n_steps, initial_condition=ic)
 
         return ks_time_series
+
+    def predict(self, n_steps: int, initial_condition: np.ndarray):
+        if initial_condition.size != self.spatial_downsampling_factor * self.N:
+            # upsample initial condition array by spatial_downsampling_factor
+            ic_x = range(initial_condition.size)
+            interp_ic = interp1d(ic_x, initial_condition)
+            initial_condition = interp_ic(
+                np.linspace(
+                    ic_x[0], ic_x[-1], self.N * self.spatial_downsampling_factor
+                )
+            )
```

Comment:
I might handle this interpolation explicitly within `create_imperfect_prediction_train_data` and maybe instead raise an error here if the initial condition size does not match the appropriate size based on the KS parameters.  Handling interpolation automatically is convenient, but I think it could also mask potentially unintended bugs.